pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -71,13 +86,26 @@ func (f Fingerprint) Equal(other *Fingerprint) bool {
 // a fingerprint. As the file grows, its fingerprint is updated
 // until it reaches a maximum size, as configured on the operator
 func (f Fingerprint) StartsWith(old *Fingerprint) bool {
-	l0 := len(old.FirstBytes)
-	if l0 == 0 {
+	lenOld := old.BytesLength
+	if lenOld == 0 {
 		return false
 	}
-	l1 := len(f.FirstBytes)
-	if l0 > l1 {
+	lenCurrent := len(f.firstBytes)
+	if lenOld > lenCurrent {
 		return false
 	}
-	return bytes.Equal(old.FirstBytes[:l0], f.FirstBytes[:l0])
+	h := fnv.New128a()
+	h.Write(f.firstBytes[:lenOld])
+	hash := h.Sum(nil)
+	return bytes.Equal(old.HashBytes, hash)
+}
+
+// IsCompleteFingerprint checks to see if fingerprint has room for updates
+func (f Fingerprint) IsCompleteFingerprint(maxFingerprintSize int, offset int64) bool {
```

Comment:
```suggestion
func (f Fingerprint) IsCompleteFingerprint(maxFingerprintSize int, offset int64) bool {
```
Can we choose a better name and comments? `IsMaxSize`?